### PR TITLE
Use LaTeX bibliography in step-14

### PR DESCRIPTION
**Pull request by adam4130**
In issue #8152 it was suggested to convert the steps to LaTeX style bibliographies. This request demonstrates this for step-14. It produces a clean output, however, it requires one to have `bibtex` in their path and puts all reference entries into a single file. Before I convert the rest of the steps, are these trade-offs worth it?

### CODE REVIEW

File: doc/doxygen/references.bib [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  journal = {Acta Numerica}, 
+  year =    2001, 
+  volume =  10, 
+  pages =   {1--102}
+}
+
+@phdthesis{Bec95,
+  author =  {Roland Becker},
+  title =   {An Adaptive Finite Element Method for the Incompressible 
+             Navier-Stokes Equations on Time-dependent Domains},
```
Comment (bangerth) [inline doc/doxygen/references.bib]: You'll have to put `{...}` around the capitals in "Navier" and "Stokes".

File: doc/doxygen/references.bib [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+             Basic Analysis and Examples},
+  journal = {East-West J. Numer. Math},
+  volume =  4,
+  year =    1996,
+  pages =   {237--264}
+}
+
+@inproceedings{BR95,
+  author =  {Roland Becker and Rolf Rannacher},
+  title =   {Weighted A Posteriori Error Control in FE Methods},
```
Comment (bangerth) [inline doc/doxygen/references.bib]: and here in `{FE}`

**Comment by jppelteret** [comment]
@bangerth Are you happy with the changes made here?

File: doc/doxygen/references.bib (hunk 1 above)
Comment (bangerth) [reply doc/doxygen/references.bib]: My comment still applies here.

File: doc/doxygen/references.bib [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  title =   {An Adaptive Finite Element Method for the Incompressible 
+             Navier-Stokes Equations on Time-dependent Domains},
+  school =  {Universit{\"a}t Heidelberg},
+  type =    {Dissertation},
+  year =    1995
+}
+
+@techreport{Bec98,
+  author =  {Roland Becker},
+  title =   {Weighted Error Estimators for the Incompressible {N}avier-{S}tokes Equations},
```
Comment (bangerth) [inline doc/doxygen/references.bib]: But you did it correctly here.

File: doc/doxygen/references.bib [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+@phdthesis{Har02,
+  author =  {Ralf Hartmann},
+  title =   {Adaptive Finite Element Methods for the Compressible Euler Equations},
+  school =  {Universit{\"a}t Heidelberg},
+  year =    2002
+}
+
+@article{HH01,
+  author =  {Ralf Hartmann and Paul Houston},
+  title =   {Adaptive Discontinuous Galerkin Finite Element Methods for
```
Comment (bangerth) [inline doc/doxygen/references.bib]: Also braces around the `G` in Galerkin here.

File: doc/doxygen/references.bib [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  year =    2003,
+  volume =  24,
+  number =  3,
+  pages =   {979--1004}
+}
+
+@article{HH01b,
+  author =  {Ralf Hartmann and Paul Houston},
+  title =   {Adaptive Discontinuous Galerkin Finite Element Methods for the 
+             Compressible Euler Equations},
```
Comment (bangerth) [inline doc/doxygen/references.bib]: And the E in Euler

**Comment by bangerth** [comment]
Thanks, @adam4130 ! In the future, when you address requested changes, add a note that you did so -- we don't get informed just because you push a new commit.

Do you want to look at some of the other tutorial programs as well?

**Comment by adam4130** [comment]
I will start leaving notes from now on. In the next few days I'll convert the rest of the tutorials over.

**Comment by bangerth** [comment]
Great, thanks in advance!

**Pull request merged**